pkg/types/user.go

```diff
@@ -1,42 +1,216 @@
 )
 
-type Invite struct {
-	bun.BaseModel `bun:"table:user_invite"`
+var (
+	ErrUserAlreadyExists               = errors.MustNewCode("user_already_exists")
+	ErrPasswordAlreadyExists           = errors.MustNewCode("password_already_exists")
+	ErrUserNotFound                    = errors.MustNewCode("user_not_found")
+	ErrResetPasswordTokenAlreadyExists = errors.MustNewCode("reset_password_token_already_exists")
+	ErrPasswordNotFound                = errors.MustNewCode("password_not_found")
+	ErrResetPasswordTokenNotFound      = errors.MustNewCode("reset_password_token_not_found")
+)
 
-	Identifiable
-	TimeAuditable
-	OrgID string `bun:"org_id,type:text,notnull" json:"orgId"`
-	Name  string `bun:"name,type:text,notnull" json:"name"`
-	Email string `bun:"email,type:text,notnull,unique" json:"email"`
-	Token string `bun:"token,type:text,notnull" json:"token"`
-	Role  string `bun:"role,type:text,notnull" json:"role"`
-}
+type UserStore interface {
+	// invite
+	CreateBulkInvite(ctx context.Context, invites []*Invite) error
+	ListInvite(ctx context.Context, orgID string) ([]*Invite, error)
+	DeleteInvite(ctx context.Context, orgID string, id valuer.UUID) error
+	GetInviteByToken(ctx context.Context, token string) (*Invite, error)
+	GetInviteByEmailInOrg(ctx context.Context, orgID string, email string) (*Invite, error)
+
+	// user
+	CreateUserWithPassword(ctx context.Context, user *User, password *FactorPassword) (*User, error)
+	CreateUser(ctx context.Context, user *User) error
+	GetUserByID(ctx context.Context, orgID string, id string) (*User, error)
+	GetUserByEmailInOrg(ctx context.Context, orgID string, email string) (*User, error)
+	GetUsersByEmail(ctx context.Context, email string) ([]*User, error)
+	ListUsers(ctx context.Context, orgID string) ([]*User, error)
+	UpdateUser(ctx context.Context, orgID string, id string, user *User) (*User, error)
+	DeleteUser(ctx context.Context, orgID string, id string) error
 
-type GettableUser struct {
-	User
-	Organization string `json:"organization"`
+	// password
+	CreateResetPasswordToken(ctx context.Context, resetPasswordRequest *FactorResetPasswordRequest) error
+	GetPasswordByUserID(ctx context.Context, id string) (*FactorPassword, error)
+	GetFactorResetPassword(ctx context.Context, token string) (*FactorResetPasswordRequest, error)
+	UpdatePassword(ctx context.Context, userID string, password string) error
+	UpdatePasswordAndDeleteResetPasswordEntry(ctx context.Context, userID string, password string) error
 }
 
 type User struct {
-	bun.BaseModel `bun:"table:users"`
+	bun.BaseModel `bun:"table:user"`
 
+	Identifiable
 	TimeAuditable
-	ID                string `bun:"id,pk,type:text" json:"id"`
-	Name              string `bun:"name,type:text,notnull" json:"name"`
-	Email             string `bun:"email,type:text,notnull,unique" json:"email"`
-	Password          string `bun:"password,type:text,notnull" json:"-"`
+	HName             string `bun:"h_name,type:text,notnull" json:"hName"`
```

Comment:
```suggestion
	DisplayName             string `bun:"display_name,type:text,notnull" json:"displayName"`
```

and let's get rid of ProfilePictureURL as well.

Reply:
done